.github/workflows/run-core-traits-tests.yml

```diff
@@ -21,6 +21,7 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
+        allow-prereleases: true
```

Comment:
This allows setup-python to install a Python 3.12 pre-release, and avoids the need to change these workflows _again_ once Python 3.12.0 is released.